pulsar/producer_partition.go

```diff
@@ -126,6 +129,23 @@ func newPartitionProducer(client *client, topic string, options *ProducerOptions
 		p.producerName = options.Name
 	}
 
+	encryption := options.Encryption
+	// add default message crypto if not provided
+	if encryption != nil && len(encryption.Keys) > 0 && encryption.MessageCrypto == nil {
+		logCtx := fmt.Sprintf("[%v] [%v] [%v]", p.topic, p.producerName, p.producerID)
+		messageCrypto, err := crypto.NewDefaultMessageCrypto(logCtx, true, logger)
+		if err != nil {
+			logger.WithError(err).Error("Unable to get MessageCrypto instance. Producer creation is abandoned")
```

Comment:
Will there be more context in the err of why this failed?

Reply:
Yes. Error is returned if generation of data key is failed. Here we are logging the same error with other information.